web/src/message_fetch.ts

```diff
@@ -226,7 +226,7 @@ function get_messages_success(data: MessageFetchResponse, opts: MessageFetchOpti
 
     if (
         opts.msg_list &&
-        !opts.msg_list.is_combined_feed_view &&
+        !opts.msg_list.should_preserve_current_rendered_state() &&
         opts.msg_list !== message_lists.current
     ) {
         // We unnarrowed before receiving new messages so
```

Comment:
Hmm, but I guess this logic may not be correct, in that we could process them for a `MessageListData`? It seems bad to do `finish_newer_batch` without actually adding the data to a `MessageListData` object.

Reply:
Ahh, indeed, that's fixed in the second commit :)

Reply:
Yup